test/sequence_number.cpp

```diff
@@ -131,14 +134,14 @@ BOOST_AUTO_TEST_CASE(medium_distances)
     BOOST_CHECK_EQUAL(largest.distance(second_largest), -1);
     BOOST_CHECK_EQUAL(largest.distance(zero), 1);
     BOOST_CHECK_EQUAL(zero.distance(largest), -1);
-    BOOST_CHECK_EQUAL(zero.distance(half), half.value());
-    BOOST_CHECK_EQUAL(half.distance(zero), -half.value());
-    BOOST_CHECK_EQUAL(one.distance(half), half.value() - 1);
-    BOOST_CHECK_EQUAL(half.distance(one), -(half.value() - 1));
-    BOOST_CHECK_EQUAL(largest.distance(half), -(half.value() + 1));
+    BOOST_CHECK_EQUAL(zero.distance(half), static_cast<difference_type>(half.value()));
```

Comment:
Note to myself: consider a specialized `value<difference_type>()` that returns `difference_type` to avoid explicit `static_cast`.